move/axelar_gateway/sources/versioned/gateway_v0.move

```diff
@@ -0,0 +1,485 @@
+        source_chain,
+        message_id,
+        source_address,
+        destination_id,
+        bytes32::from_bytes(hash::keccak256(&payload)),
+    );
+
+    assert!(
+        self[command_id] == message_status::approved(message.hash()),
+        EMessageNotApproved,
+    );
+
+    let message_status_ref = &mut self[command_id];
+    *message_status_ref = message_status::executed();
+
+    events::message_executed(
+        message,
+    );
+
+    // Friend only.
+    channel::create_approved_message(
+        source_chain,
+        message_id,
+        source_address,
+        destination_id,
+        payload,
+    )
+}
+
+public(package) fun send_message(_self: &GatewayV0, message: MessageTicket, current_version: u64) {
+    let (
+        source_id,
+        destination_chain,
+        destination_address,
+        payload,
+        version,
+    ) = message.destroy();
+
+    assert!(version <= current_version, ENewerMessage);
+
+    events::contract_call(
+        source_id,
+        destination_chain,
+        destination_address,
+        payload,
+        address::from_bytes(hash::keccak256(&payload)),
+    );
+}
+
+// -----------------
+// Private Functions
+// -----------------
+
+fun peel_messages(message_data: vector<u8>): vector<Message> {
+    utils::peel!(
+        message_data,
+        |bcs| {
+            let messages = vector::tabulate!(
+                bcs.peel_vec_length(),
+                |_| message::peel(bcs),
+            );
+            assert!(messages.length() > 0, EZeroMessages);
+            messages
+        },
+    )
+}
+
+fun data_hash(command_type: u8, data: vector<u8>): Bytes32 {
+    let mut typed_data = vector::singleton(command_type);
+    typed_data.append(data);
+
+    bytes32::from_bytes(hash::keccak256(&typed_data))
+}
+
+fun approve_message(self: &mut GatewayV0, message: message::Message) {
+    let command_id = message.command_id();
+
+    // If the message was already approved, ignore it.
+    if (self.messages.contains(command_id)) {
+        return
+    };
+
+    self
+        .messages
+        .add(
+            command_id,
+            message_status::approved(message.hash()),
+        );
+
+    events::message_approved(
+        message,
+    );
+}
+
+/// ---------
+/// Test Only
+/// ---------
+#[test_only]
+public(package) fun operator(self: &GatewayV0): &address {
+    &self.operator
+}
+
+#[test_only]
+public(package) fun operator_mut(self: &mut GatewayV0): &mut address {
+    &mut self.operator
+}
+
+#[test_only]
+public(package) fun messages(self: &GatewayV0): &Table<Bytes32, MessageStatus> {
+    &self.messages
+}
+
+#[test_only]
+public(package) fun messages_mut(
+    self: &mut GatewayV0,
+): &mut Table<Bytes32, MessageStatus> {
+    &mut self.messages
+}
+
+#[test_only]
+public(package) fun signers(self: &GatewayV0): &AxelarSigners {
+    &self.signers
+}
+
+#[test_only]
+public(package) fun signers_mut(self: &mut GatewayV0): &mut AxelarSigners {
```

Comment:
i don't see this being used anywhere. can you double check other getters as well? And if they're only used within this module, then convert them to internal functions